test/framework/config.py

```diff
@@ -428,6 +428,96 @@ def test_XDG_CONFIG_env_vars(self):
+        orig_sys_path = sys.path[:]
+        sys.path = [tmpdir] + [p for p in sys.path if not os.path.exists(os.path.join(p, 'easybuild', 'easyconfigs'))]
+
+        # default: only pick up installed easyconfigs via sys.path
+        eb_go = eboptions.parse_options(args=[])
+        self.assertEqual(eb_go.options.robot_paths, [tmp_ecs_dir])
+
+        # prepend to default robot path
+        eb_go = eboptions.parse_options(args=['--robot-paths=/foo:'])
+        self.assertEqual(eb_go.options.robot_paths, ['/foo', tmp_ecs_dir])
+        eb_go = eboptions.parse_options(args=['--robot-paths=/foo:/bar/baz/:'])
+        self.assertEqual(eb_go.options.robot_paths, ['/foo', '/bar/baz/', tmp_ecs_dir])
+
+        # append to default robot path
+        eb_go = eboptions.parse_options(args=['--robot-paths=:/bar/baz'])
+        self.assertEqual(eb_go.options.robot_paths, [tmp_ecs_dir, '/bar/baz'])
+        # append to default robot path
+        eb_go = eboptions.parse_options(args=['--robot-paths=:/bar/baz:/foo'])
+        self.assertEqual(eb_go.options.robot_paths, [tmp_ecs_dir, '/bar/baz', '/foo'])
+
+        # prepend and append to default robot path
+        eb_go = eboptions.parse_options(args=['--robot-paths=/foo/bar::/baz'])
+        self.assertEqual(eb_go.options.robot_paths, ['/foo/bar', tmp_ecs_dir, '/baz'])
+        eb_go = eboptions.parse_options(args=['--robot-paths=/foo/bar::/baz:/trala'])
+        self.assertEqual(eb_go.options.robot_paths, ['/foo/bar', tmp_ecs_dir, '/baz', '/trala'])
+        eb_go = eboptions.parse_options(args=['--robot-paths=/foo/bar:/trala::/baz'])
+        self.assertEqual(eb_go.options.robot_paths, ['/foo/bar', '/trala', tmp_ecs_dir, '/baz'])
+
+        # also via $EASYBUILD_ROBOT_PATHS
+        os.environ['EASYBUILD_ROBOT_PATHS'] = '/foo::/bar/baz'
+        eb_go = eboptions.parse_options(args=[])
+        self.assertEqual(eb_go.options.robot_paths, ['/foo', tmp_ecs_dir, '/bar/baz'])
+
+        # combining $EASYBUILD_ROBOT_PATHS and --robot-paths: all paths are retained in the order to be expected
+        os.environ['EASYBUILD_ROBOT_PATHS'] = '/foo::/bar/baz'
+        eb_go = eboptions.parse_options(args=['--robot-paths=/one::/last'])
+        self.assertEqual(eb_go.options.robot_paths, ['/one', '/foo', tmp_ecs_dir, '/bar/baz', '/last'])
+
+        del os.environ['EASYBUILD_ROBOT_PATHS']
+
+        # also works with a cfgfile in the mix
+        config_file = os.path.join(self.tmpdir, 'testconfig.cfg')
+        cfgtxt = '\n'.join([
+            '[config]',
+            'robot-paths=/cfgfirst::/cfglast',
+        ])
+        write_file(config_file, cfgtxt)
+        eb_go = eboptions.parse_options(args=['--configfiles=%s' % config_file])
+        self.assertEqual(eb_go.options.robot_paths, ['/cfgfirst', tmp_ecs_dir, '/cfglast'])
+
+        # cfgfile entry is lost when env var and/or cmdline options are used
```

Comment:
Of course, but then why doesn't command line override environment, but complement it?

```
export EASYBUILD_ROBOT_PATHS='/foo::/bar/baz'
eb --robot-paths /one::/last
```

gives `['/one', '/foo', <DEFAULT>, '/bar/baz', '/last']` as robot search path instead of `['/one', <DEFAULT>, '/last']` (see test above). Hence my confusion...